### PR TITLE
[TK-3]: intergrate analytics and crashlytics 

### PR DESCRIPTION
**Pull request by ammadahmed22**
## Jira [TK-3](https://shop-go.atlassian.net/jira/software/projects/TK/boards/2?selectedIssue=TK-3)

**Description**:
- Integrated Analytics
- Integrated Crashlytics
- Crash and touch of a button are logged into firebase
- User locations are also logged

**How to test it?**
- Add FatalError() in viewDidLoad() once the app crashes pause the app and reopen the app on stimulator.
- Add a button with let numbers = [0] let _ = numbers[1] run the app, stop the app, open sim click the button and run again
- Test Analytics add a button with Analytics.logEvent("button_pressed", parameters: nil)

### CODE REVIEW

File: Vape-X/ViewController.swift [hunk 1]
```diff
@@ -6,14 +6,20 @@
 //
 
 import UIKit
+import FirebaseAnalytics
+
```
Comment (haroonmaqsoodch) [inline Vape-X/ViewController.swift:10]: remove extra space

File: Vape-XTests/Vape_XTests.swift [hunk 2]
```diff
@@ -6,6 +6,8 @@
 //
 
 import XCTest
+import FirebaseCore
+//import FirebaseCore
```
Comment (haroonmaqsoodch) [inline Vape-XTests/Vape_XTests.swift:10]: delete this comment

File: Vape-X/ViewController.swift [hunk 3]
```diff
@@ -6,14 +6,19 @@
...
 class ViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
+
     }
 
+    
+
```
Comment (haroonmaqsoodch) [inline Vape-X/ViewController.swift:20]: remove the space here

File: Vape-X/AppDelegate.swift [hunk 4]
```diff
@@ -6,17 +6,22 @@
 //
 
 import UIKit
+import FirebaseCore
 
-@main
-class AppDelegate: UIResponder, UIApplicationDelegate {
```
Comment (haroonmaqsoodch) [inline Vape-X/AppDelegate.swift:11]: remove one space from here as well

File: Vape-X/AppDelegate.swift [hunk 5]
```diff
@@ -6,17 +6,22 @@
...
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
 
+  var window: UIWindow?
 
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
-        return true
-    }
+  func application(_ application: UIApplication,
```
Comment (haroonmaqsoodch) [inline Vape-X/AppDelegate.swift:17]: this whole method is not indented properly, please fix

File: Vape-X/ViewController.swift (hunk 1 above)
Comment (ammadahmed22) [reply Vape-X/ViewController.swift:10]: fixed

File: Vape-XTests/Vape_XTests.swift (hunk 2 above)
Comment (ammadahmed22) [reply Vape-XTests/Vape_XTests.swift]: fixed

File: Vape-X/ViewController.swift (hunk 3 above)
Comment (ammadahmed22) [reply Vape-X/ViewController.swift]: fixed

File: Vape-X/AppDelegate.swift (hunk 4 above)
Comment (ammadahmed22) [reply Vape-X/AppDelegate.swift]: fixed

File: Vape-X/AppDelegate.swift (hunk 5 above)
Comment (ammadahmed22) [reply Vape-X/AppDelegate.swift]: fixed

**Review by haroonmaqsoodch — APPROVED** [review]
LGTM ✅

File: Vape-X.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -437,7 +502,8 @@
 				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
-				DEVELOPMENT_TEAM = 96839QFWQR;
```
[the remark this answers is not in the document]
Comment (ammadahmed22) [reply Vape-X.xcodeproj/project.pbxproj:440]: also change of the dev name, before I didn't have a team set and later changed it to my personal team.

File: Vape-X/AppDelegate.swift [hunk 7]
```diff
@@ -6,17 +6,22 @@
...
+    didFinishLaunchingWithOptions launchOptions:
+                   [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    FirebaseApp.configure() // To use Firebase
```
[the remark this answers is not in the document]
Comment (ammadahmed22) [reply Vape-X/AppDelegate.swift]: This is on github side it messing up the code in project file this how code looks like in project file image

File: Vape-X.xcodeproj/project.pbxproj (hunk 6 above)
[the remark this answers is not in the document]
Comment (ammadahmed22) [reply Vape-X.xcodeproj/project.pbxproj:440]: fix

**Review by MuhammadHuzaifa2 — APPROVED** [review]

**Review by basitalich — APPROVED** [review]

**Pull request merged**